ViMac-Swift/UserPreferences.swift

```diff
@@ -75,45 +75,50 @@ struct UserPreferences {
             typealias T = String
             
             static var key = "ScrollCharacters"
-            static var defaultValue = "hjkldu"
+            static var defaultValue = "h,j,k,l,d,u,G,gg"
             
             static func isValid(value keys: String) -> Bool {
-                let isCountValid = keys.count == 4 || keys.count == 6
-                let areKeysUnique = keys.count == Set(keys).count
-                return isCountValid && areKeysUnique
+                let keySequences = keys.components(separatedBy: ",")
+                let isCountValid = keySequences.count == 4 || keySequences.count == 6 || keySequences.count == 8
+                let areKeySequencesUnique = keySequences.count == Set(keySequences).count
+                return isCountValid && areKeySequencesUnique
             }
             
             static func readAsConfig() -> ScrollKeyConfig {
                 let s = read()
-                
-                let scrollLeftKey = s[s.index(s.startIndex, offsetBy: 0)]
-                let scrollDownKey = s[s.index(s.startIndex, offsetBy: 1)]
-                let scrollUpKey = s[s.index(s.startIndex, offsetBy: 2)]
-                let scrollRightKey = s[s.index(s.startIndex, offsetBy: 3)]
-                let scrollHalfDownKey = s.count == 6 ? (s[s.index(s.startIndex, offsetBy: 4)]) : nil
-                let scrollHalfUpKey = s.count == 6 ? (s[s.index(s.startIndex, offsetBy: 5)]) : nil
+                let keySequences = s.components(separatedBy: ",")
+                let scrollLeftKey = keySequences[0]
+                let scrollDownKey = keySequences[1]
+                let scrollUpKey = keySequences[2]
+                let scrollRightKey = keySequences[3]
+                let scrollHalfDownKey = keySequences.count >= 6 ? (keySequences[4]) : nil
+                let scrollHalfUpKey = keySequences.count >= 6 ? (keySequences[5]) : nil
+                let scrollBottomKey = keySequences[6]
+                let scrollTopKey = keySequences[7]
```

Comment:
This will crash since there is no length check.

```
let scrollBottomKey = keySequences.count >= 8 ? keySequences[6] : nil
let scrollTopKey = keySequences.count >= 8 ? keySequences[7] : nil
```